### PR TITLE
Rename selector to partialfilterselector in indexes

### PR DESCRIPTION
**Pull request by garrensmith**
To make it easier to distinguish between a selector in _find and a
selector in _index. Rename the selector in the _index to
partialfilterselector. It also gives a bit more of an explanation of
what this selector does.

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

**Comment by willholley** [comment]
It needs rebasing on master but basically looks good to me. What happens if both `selector` and `partialfilterselector` are specified?

**Comment by garrensmith** [comment]
@willholley I've just rebased

**Comment by willholley** [comment]
@garrensmith I think we should follow the convention for other fields and use `_` as a delimiter - i.e. `partial_filter_selector` instead of `partialfilterselector`.

File: src/mango/src/mango_cursor.erl [hunk 1]
```diff
@@ -92,7 +92,7 @@ maybe_filter_indexes_by_ddoc(Indexes, Opts) ->
         {use_index, []} ->
             %We remove any indexes that have a selector
```
Comment (willholley) [inline src/mango/src/mango_cursor.erl]: probably add a space here

File: src/mango/src/mango_idx.erl [hunk 2]
```diff
@@ -44,7 +44,7 @@
     to_json/1,
     delete/4,
     get_usable_indexes/3,
-    get_idx_selector/1
+    get_idx_partial_filter_selector/1
```
Comment (willholley) [inline src/mango/src/mango_idx.erl]: rename to `get_partial_filter_selector` - that it is in the `mango_idx` module implies that it relates to an index

File: src/mango/src/mango_idx.erl [hunk 3]
```diff
@@ -139,7 +139,25 @@ new(Db, Opts) ->
 
 validate_new(Idx, Db) ->
     Mod = idx_mod(Idx),
-    Mod:validate_new(Idx, Db).
+    Def1 = maybe_rename_idx_fields(Idx#idx.def),
+    Idx1 = Idx#idx{def=Def1},
+    Mod:validate_new(Idx1, Db).
+
+
+maybe_rename_idx_fields({Props}) ->
```
Comment (willholley) [inline src/mango/src/mango_idx.erl]: rename to `maybe_rename_fields`

File: src/mango/test/03-operator-test.py [hunk 4]
```diff
@@ -185,7 +185,7 @@ def test_lt_includes_null_but_not_missing(self):
 
     def test_lte_includes_null_but_not_missing(self):
         docs = self.db.find({
-                "twitter": {"$lt": 1}
+                "twitter": {"$lte": 1}
```
Comment (willholley) [inline src/mango/test/03-operator-test.py]: move this into a separate PR as it's unrelated

**Comment by willholley** [comment]
after squashing, looks ok to me

**Comment by garrensmith** [comment]
@davisp I've made all the required changes. Could you take a look when you get a chance.

File: src/mango/src/mango_idx_view.erl [hunk 5]
```diff
@@ -199,8 +199,8 @@ opts() ->
             {tag, fields},
             {validator, fun mango_opts:validate_sort/1}
         ]},
-        {<<"selector">>, [
-            {tag, selector},
```
Comment (davisp) [inline src/mango/src/mango_idx_view.erl]: I see the get_legacy_selector code above, but reading through code I can't convince myself that removing this option definition won't break old indexes that used the selector name. I wonder if we shouldn't continue to support this for awhile and upgrade it internally for new indexes?

Comment (davisp) [reply src/mango/src/mango_idx_view.erl]: And/or perhaps just adding a test specifically for the upgrade would be enough here. And for that I'd like to see us write an old style ddoc directly and then read the index to make sure it behaves as expected.
